[CI:DOCS] Update basic_networking.md to include a 2nd Example for Rootless Networking

reo-1963 opened the pull request:
If creating a network is going to have no "network_dns_servers" by default, you guys should describe how to make this work in detail so noobs like me don't start thinking we ought to go back to Docker.  By all means have a real Tech Writer and a real Network Engineer review my proposed change. #18530

Richard Oberuc

#### Does this PR introduce a user-facing change?

None

baude committed:
enabled hyperv image downloads

now that we have public hypverv fcos artifacts, we can download them
instead of requiring a special build.

Signed-off-by: Brent Baude <bbaude@redhat.com>

[NO NEW TESTS NEEDED]

geichelberger committed:
Add support for host-gateway

The `--add-host` option now accepts the special string `host-gateway`
instead of an IP Address, which will be mapped to the host IP address.

Signed-off-by: Gregor Eichelberger <gregor.eichelberger@tuwien.ac.at>

edsantiago committed:
New partial-line test is flaking

...possibly because we somehow ended up with a two-line
log file for a simple 'echo hi'? Make our timestamp-getting
code safer by adding 'head -1'.

Signed-off-by: Ed Santiago <santiago@redhat.com>

cevich committed:
Cirrus: Prune defunct job + fix noop alias

The mechanism fueling the benchmarks job has long since been removed.
Running the job is useless now, remove it.

Signed-off-by: Chris Evich <cevich@redhat.com>

lsm5 committed:
[CI:BUILD] Packit: add back fedora-eln targets

Fedora ELN targets were removed in the switch to ephemeral coprs. Add
them back.

Podman rpm doesn't depend on go-md2man package anymore and instead uses
vendored go-md2man for building manpages, so that's no longer a reason
for build failures on Fedora ELN.

[NO NEW TESTS NEEDED]

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

matejvasek committed:
Revert "compat,build: pull must accept string"

This reverts commit 5b148a0a68360892f57603b77e73c69e32fb7b7e.

Reverting to treating the `pull` query parameter as a boolean.
Because of deceiving Docker API documentation it was assumed that the
parameter is pull-policy, however that is not true. Docker does treat
`pull` as a boolean. What is interesting is that Docker indeed accepts
strings like `always` or `never` however Docekr both of these strings
treat as `true`, not as pull-policy. As matter of the fact it seems
there is no such a thing as pull-policy in Docker.

More context https://github.com/containers/podman/issues/17778#issuecomment-1673931925

Signed-off-by: Matej Vasek <mvasek@redhat.com>

matejvasek committed:
fix: Docker API compatible bool deserialization

In Docker anything but "", "0", "no", "false", "none" (ignoring case) is considered to be true.

Signed-off-by: Matej Vasek <mvasek@redhat.com>

Luap99 committed:
fix podman top missing output flake

Sometimes there is no output displayed from the podman top command but
no error is shown either. Looking at the code I think the issue here is
that we do not wait for the output reader to end as it runs in a
different goroutine. Thus the last lines of output might be missing.

The fix is simply to wait for said goroutine to finish before returning.
While at it also fix the missing scanner error check and return the read
errors back to the caller.

[NO NEW TESTS NEEDED] It is a flake.

Fixes #19504

Signed-off-by: Paul Holzinger <pholzing@redhat.com>

renovate[bot] committed:
chore(deps): update dependency golangci/golangci-lint to v1.54.1

Signed-off-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ashley-cui committed:
[CI:DOCS] Update Release Notes and Release Process

Signed-off-by: Ashley Cui <acui@redhat.com>

reo-1963 committed:
Update basic_networking.md

If creating a network is going to have no "network_dns_servers" by default, you guys should describe how to make this work in detail so noobs like me don't start thinking we ought to go back to Docker. 
By all means have a real Tech Writer and a real Network Engineer review my proposed change. #18530

Signed-off-by: reo-1963 <107125510+reo-1963@users.noreply.github.com>

openshift-merge-robot committed:
Merge pull request #19589 from lsm5/main-packit

[CI:BUILD] Packit: add back fedora-eln targets

timway commented:
I'm not sure I understand this example. Looking at Fedora running rootless containers this is what I built to model what you are doing:

```bash
#! /usr/bin/bash

LOOKUP="nslookup -type=aaaa quad9.net"

CREATED_NETWORK=0
if ! podman network ls --filter name=testing --noheading | grep testing > /dev/null
then
    CREATED_NETWORK=1
    podman network create testing > /dev/null
fi

for TYPE in default podman testing
do
    NETWORK=""
    if [ "$TYPE" != "default" ]
    then
        NETWORK="--network $TYPE"
    fi

    echo "-*-*-*- Testing: $TYPE -*-*-*-"
    ( set -x ; podman run --detach --name dns-rootless-net-$TYPE $NETWORK --quiet --rm registry.access.redhat.com/ubi9/ubi-init > /dev/null)
    podman exec dns-rootless-net-$TYPE dnf --assumeyes install bind-utils ncurses > /dev/null
    echo $LOOKUP
    ( set -x ; podman exec --tty dns-rootless-net-$TYPE $LOOKUP | \
        python3 -c "import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\n')")
    ( set -x ; podman inspect dns-rootless-net-$TYPE | jq .[].HostConfig.NetworkMode)
    podman stop dns-rootless-net-$TYPE > /dev/null

    if [ "$TYPE" != "testing" ]
    then
        echo
        echo
    fi
done

if [ $CREATED_NETWORK != 0 ]
then
    podman network rm testing > /dev/null
fi
```

It results in:

```bash
$ bash blah.sh 
-*-*-*- Testing: default -*-*-*-
+ podman run --detach --name dns-rootless-net-default --quiet --rm registry.access.redhat.com/ubi9/ubi-init
nslookup -type=aaaa quad9.net
+ podman exec --tty dns-rootless-net-default nslookup -type=aaaa quad9.net
+ python3 -c 'import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\''\n'\'')'
Server:         10.0.2.3
Address:        10.0.2.3#53

Non-authoritative answer:
Name:   quad9.net
Address: 2620:0:871:9000::77
+ podman inspect dns-rootless-net-default
+ jq '.[].HostConfig.NetworkMode'
"slirp4netns"


-*-*-*- Testing: podman -*-*-*-
+ podman run --detach --name dns-rootless-net-podman --network podman --quiet --rm registry.access.redhat.com/ubi9/ubi-init
nslookup -type=aaaa quad9.net
+ podman exec --tty dns-rootless-net-podman nslookup -type=aaaa quad9.net
+ python3 -c 'import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\''\n'\'')'
Server:         10.211.11.254
Address:        10.211.11.254#53

Non-authoritative answer:
Name:   quad9.net
Address: 2620:0:871:9000::77
+ podman inspect dns-rootless-net-podman
+ jq '.[].HostConfig.NetworkMode'
"bridge"


-*-*-*- Testing: testing -*-*-*-
+ podman run --detach --name dns-rootless-net-testing --network testing --quiet --rm registry.access.redhat.com/ubi9/ubi-init
nslookup -type=aaaa quad9.net
+ podman exec --tty dns-rootless-net-testing nslookup -type=aaaa quad9.net
+ python3 -c 'import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\''\n'\'')'
Server:         10.89.1.1
Address:        10.89.1.1#53

Non-authoritative answer:
Name:   quad9.net
Address: 2620:0:871:9000::77
+ podman inspect dns-rootless-net-testing
+ jq '.[].HostConfig.NetworkMode'
"bridge"
```

TLDR; I don't observe the broken DNS stack in any case. Is this just an issue in Podman machines or possibly in WSL2? Limited to VPN or similar conditions?

Also this PR is against your fork not the main repo like I think you intended to do.

wjt committed:
Add missing verb in machinectl example

Without the verb 'shell', the invocation fails with:

    Unknown command verb testuser@

Signed-off-by: Will Thompson <will@willthompson.co.uk>

klesh commented:
Seems like this PR is targeting your own `main` branch. Shouldn't it be the [upstream](https://github.com/containers/podman) instead?
![image](https://github.com/reo-1963/podman/assets/61080/6ac49582-3a33-43e6-9848-b928b9e61507)

openshift-merge-robot committed:
Merge pull request #19599 from ashley-cui/rel

[CI:DOCS] Update Release Notes and Release Process

openshift-merge-robot committed:
Merge pull request #19607 from wjt/performance-machinectl-shell

[CI:DOCS] Add missing verb in machinectl example

openshift-merge-robot committed:
Merge pull request #19598 from containers/renovate/golangci-golangci-lint-1.x

chore(deps): update dependency golangci/golangci-lint to v1.54.1

openshift-merge-robot committed:
Merge pull request #19152 from geichelberger/add-host-gateway-support

Add support for host-gateway

openshift-merge-robot committed:
Merge pull request #19587 from cevich/kill_bench_job

[CI:BUILD] Cirrus: Prune defunct job + fix noop alias

openshift-merge-robot committed:
Merge pull request #19576 from edsantiago/logpartial_flake

New partial-line test is flaking

PeterWhittaker committed:
Update rootfs.md: Fix formatting and wording of idmap option

The newly introduced `idmap` section of rootfs lacked a header
(comparable to Overlay Rootfs Mounts), had odd formatting, and
wording that differed from other instances of idmap, e.g., the
one in the --volume section. This commits addresses those issues.

Signed-off-by: Peter Whittaker <PeterWhittaker@SphyrnaSecurity.com>

reo-1963 commented:
I'm not sure about where the "pull" or "merge" goes. My "Git" skills are not very good.

The problem with setting the DNS on a "network create" is very specific to just happening on WSL2 on Windows. #18530.
But if WSL2 is how Win 10 & Win 11 are going to install Podman (and will be an issue going forward), and #18530 has been an "unresolved" issue for the past 3 months, and it was a problem for me as a "noob user", it's going to be a problem for a lot of other new Users who may decide it's a deal breaker and prematurely walk away from Podman.

Maybe this should go into a "TroubleShooting" for "WSL2 and Windows and Networking and Rootless"? 
But not finding one of those, I am suggesting a 2nd example under "Networking and Rootless".

openshift-merge-robot committed:
Merge pull request #19595 from Luap99/top-output-flake

fix podman top missing output flake

edsantiago committed:
sphinx: skip options include dir

Tell sphinx not to process the "options" dir, those files
are not to be published.

Context: websearching for podman volume info, I stumbled upon:

   https://docs.podman.io/en/latest/markdown/options/volume.html

...and panicked because I saw '<<container|pod>>', the options
include-file syntax that should never be seen by users. I thought
the filter script was broken.

Signed-off-by: Ed Santiago <santiago@redhat.com>

openshift-merge-robot committed:
Merge pull request #19613 from PeterWhittaker/main

Update rootfs.md: Fix formatting and wording of idmap option

edsantiago committed:
CI: e2e manifest_test: use image from quay

Flake suppression: going with the one-basket model of egg storage,
switch manifest_test to use an image on quay.io (was: k8s.io).

Closes: #19148

Signed-off-by: Ed Santiago <santiago@redhat.com>

cevich committed:
Cirrus: Remove EC2 experimental flag

The VM-naming feature is now mainstream.
Ref: https://github.com/cirruslabs/cirrus-ci-docs/ issue 1051

Signed-off-by: Chris Evich <cevich@redhat.com>

openshift-merge-robot committed:
Merge pull request #19591 from matejvasek/fix-build-pull-compat

Fix pull parameter parsing for compat `/build` endpoint

openshift-merge-robot committed:
Merge pull request #19617 from edsantiago/sphinx_ignore

[CI:DOCS] sphinx: skip options include dir

openshift-merge-robot committed:
Merge pull request #19619 from cevich/remove_ec2_experimental

Cirrus: Remove EC2 experimental flag

TomSweeneyRedHat committed:
[CI:DOCS] Fix git build example in build page

The git repo that was used for the example of a git build:

  `podman build https://github.com/scollier/purpletest`

no longer exists.  Someone reached out to @rhatdan about this
and he suggested using the Podman Hello World repo.  However,
that didn't exist until a little bit ago, so I've updated
the man page with a pointer there.  That should be a lot more
stable.

[NO NEW TESTS NEEDED]
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

n1hility committed:
Update WSL backend to be compat with FCOS defaults

container/common Config was recently updated to use FCOS
default names, since Windows HyperV backend needs them,
and these are common with other providers.

Map this to values the WSL backend expects.

[NO NEW TESTS NEEDED]

Signed-off-by: Jason T. Greene <jason.greene@redhat.com>

klesh commented:
@reo-1963 It's fine, you just need to edit the PR and change the `base` to `containers/podman` so your proposal can be discussed and processed by the Podman maintainers.

renovate[bot] committed:
fix(deps): update module github.com/containers/ocicrypt to v1.1.8

Signed-off-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michalbiesek committed:
GetFcosArch add `riscv64` arch

[NO NEW TESTS NEEDED]

Signed-off-by: Michal Biesek <michalbiesek@gmail.com>

michalbiesek committed:
Add riscv64 architecture to the cross build target

[NO NEW TESTS NEEDED]

Signed-off-by: Michal Biesek <michalbiesek@gmail.com>

openshift-merge-robot committed:
Merge pull request #19621 from edsantiago/manifest_quay

CI: e2e manifest_test: use image from quay

openshift-merge-robot committed:
Merge pull request #19624 from TomSweeneyRedHat/dev/tsweeney/fixbuildmd

[CI:DOCS] Fix git build example in build page

renovate[bot] committed:
Update dependency setuptools to ~=68.1.0

Signed-off-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

openshift-merge-robot committed:
Merge pull request #19627 from containers/renovate/github.com-containers-ocicrypt-1.x

fix(deps): update module github.com/containers/ocicrypt to v1.1.8

openshift-merge-robot committed:
Merge pull request #19638 from containers/renovate/setuptools-68.x

Update dependency setuptools to ~=68.1.0

openshift-merge-robot committed:
Merge pull request #19626 from michalbiesek/feat-riscv64-support

Add riscv64 support for cross compilation

openshift-merge-robot committed:
Merge pull request #19596 from baude/allowdownload

enabled hyperv image downloads

reo-1963 committed:
Merge branch 'containers:main' into reo-1963-patch-1
